src/ssl.c

```diff
@@ -8522,6 +8522,70 @@ static WOLFSSL_EVP_PKEY* d2iGenericKey(WOLFSSL_EVP_PKEY** out,
     #endif /* !HAVE_FIPS || HAVE_FIPS_VERSION > 2 */
     #endif /* !NO_DH &&  OPENSSL_EXTRA && WOLFSSL_DH_EXTRA */
 
+    #ifdef HAVE_LIBOQS
+    {
+        int isFalcon;
+    #ifdef WOLFSSL_SMALL_STACK
+        falcon_key *falcon = (falcon_key *)MALLOC(sizeof(falcon_key), NULL,
+                                                  DYNAMIC_TYPE_FALCON);
+        if (falcon == NULL) {
+            return NULL;
+        }
+    #else
+        falcon_key falcon[1];
+    #endif
+        XMEMSET(falcon, 0, sizeof(falcon_key));
+
+        /* test if Falcon key */
+        if (priv) {
+            /* Try level 1 */
+            isFalcon = wc_falcon_init(falcon) == 0 &&
```

Comment:
Maybe call `wc_falcon_init` only once at the start?

Reply:
Sure, I can do that.